include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -394,23 +397,28 @@ auto stablePartitionIndexForDeflationArrays(const SizeType n, const ColType* typ
 // rank                   |     0     |     1     |     0     |
 // initial                | 0U| 1L| 2X| 3U| 4U| 5X| 6L| 7L| 8X|
 // index_sorted           | 0U| 1L| 3U| 4U| 6L| 7L| 2X| 5X| 8X| -> sort(non-deflated) | sort(deflated)
-// index_sorted_col_type  | 0U| 1L| 6L| 3U| 4U| 5X| 7L| 2X| 8X| -> rank0(ULLLXX) - rank1(UUX)
+// index_sorted_col_type  | 0U| 1L| 6L| 3U| 4U| 5X| 7L| 2X| 8X| -> rank0(ULLLXX) - rank1(UUX) (global indices)
+// i5_lc                  | 0U| 1L| 3L|...........| 4L| 2X| 5X| -> rank0(ULLLXX)  (local indices)
+// i5_lc                  |...........| 0U| 1U| 2X|...........| -> rank1(UUX)     (local indices)
 //
 // index_sorted_col_type can be used "locally":
 // on rank0               | 0U| 1L| 6L| --| --| --| 7L| 2X| 8X| -> ULLLXX
 // on rank1               | --| --| --| 3U| 4U| 5X| --| --| --| -> UUX
 //
+// i5_lc is just local to the current rank, and it is a view over index_sorted_col_type where indices are local.
+//
 // where U: Upper, D: Dense, L: Lower, X: Deflated
 //
 // The permutations will allow to keep the mapping between sorted eigenvalues and unsorted eigenvectors,
 // which is useful since eigenvectors are more expensive to permute, so we can keep them in their
 // initial order.
 //
-// @param types                 array[n] column type of each eigenvector after deflation (initial order)
-// @param evals                 array[n] of eigenvalues sorted as perm_sorted
-// @param perm_sorted           array[n] current -> initial (i.e. evals[i] -> types[perm_sorted[i]])
-// @param index_sorted          array[n] global(sort(non-deflated)|sort(deflated))) -> initial
-// @param index_sorted_coltype  array[n] local(sort(upper)|sort(dense)|sort(lower)|sort(deflated))) -> initial
+// @param types                 array[n]    column type of each eigenvector after deflation (initial order)
+// @param evals                 array[n]    of eigenvalues sorted as perm_sorted
+// @param perm_sorted           array[n]    current -> initial (i.e. evals[i] -> types[perm_sorted[i]])
+// @param index_sorted          array[n]    global(sort(non-deflated)|sort(deflated))) -> initial
+// @param index_sorted_coltype  array[n]    local(sort(upper)|sort(dense)|sort(lower)|sort(deflated))) -> initial
+// @param i5_lc                 array[n_lc] local(sort(upper)|sort(dense)|sort(lower)|sort(deflated))) -> initial
```

Comment:
note-to-self: specify that they are local indices, while in `index_sorted_coltype` they are global indices